packages/upload-media/src/components/provider/index.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import withRegistryProvider from './with-registry-provider';
+import { store as uploadStore } from '../../store';
+
+const MediaUploadProvider = withRegistryProvider( ( props: any ) => {
+	const { children, settings } = props;
+	const { updateSettings } = useDispatch( uploadStore );
```

Comment:
I think updateSettings should be made a private selector (in block editor, it's a mistake that it's not private, this should only be edited by updating the settings prop basically)

Reply:
Will change this in the parent PR and then update here.